File: ui/react/src/action/TableAction.ts
import SheetAction from "./SheetAction";

abstract class TableAction extends SheetAction {
    public tableName: string;

    constructor(sheetName: string, tableName: string) {
        super(sheetName);
        this.tableName = tableName;
    }

    public isAssetAction(): boolean {
        return true;
    }

    public targetAsset(): string {
        return this.tableName;
    }
}

export default TableAction;

File: core/src/main/java/com/ctrip/ferriswheel/core/action/TableAction.java
package com.ctrip.ferriswheel.core.action;

import com.ctrip.ferriswheel.common.action.Action;

public abstract class TableAction extends SheetAction implements Action {
    private String tableName;

    TableAction() {
    }

    TableAction(String sheetName, String tableName) {
        super(sheetName);
        this.tableName = tableName;
    }

    public String getTableName() {
        return tableName;
    }

    public void setTableName(String tableName) {
        this.tableName = tableName;
    }

}
